SalesforceCommerceCloud/pwa-kit
# @W-16889880 - Complete Social Login Integration: Connect Backend API to UI

# Description
Wire up Social Login API calls to the newly implemented UI. 
Login should work in 3 locations:

1. Login modal
2. Login page
3. Checkout page

We have configured **Google** and **Apple** as OOTB IDPs.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Add `commerce-sdk-react` API calls to the newly implemented UI
- Add `redirectURI` as parameter, as this will be different from the default `callback` path that has already been set in `app/components/_app-config` [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/template-retail-react-app/app/components/_app-config/index.jsx#L64)

- If there is an error with the login API call, there will be an error on the redirect page as so:
![Screenshot 2024-11-22 at 12 45 26 PM](https://github.com/user-attachments/assets/8956475b-4f47-4277-b778-f27a6bb909f9)

- If there is an error with the authorize API call, there will be an error on the login page as so:
<img width="200" alt="Screenshot 2024-12-03 at 12 23 18 PM" src="https://github.com/user-attachments/assets/47109934-8d85-43e0-8981-ce94bf69e283">


# How to Test-Drive This PR

- Navigate to https://wasatch-mrt-social-login-poc.mrt-storefront-staging.com/ where social and passwordless login have been enabled
- Open the login modal
- Verify that you can log in with both **Google** and **Apple**
  - Verify that you can log in from all 3 locations: login page (`/login`), login modal, and checkout page (`/checkout`)
  - If using the login modal or checkout page, verify that you are redirected back to the page where you initiated login

To run the e2e test:
- Point the test to the deployed environment by setting an env var like so: `export RETAIL_APP_HOME=https://wasatch-mrt-social-login-poc.mrt-storefront-staging.com`
- Run `npx playwright test --ui` from the monorepo root
- Run the test: **Registered shopper logged in through social retains persisted cart**
- Verify that the user is logged in and sees their user details in the secure account page as well as the items they added to cart as a guest
- ⚠️I have added the **NEW** Google PWA Kit account credentials into Password Manager in the **Imported Mobify - Product** folder !!⚠️

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
